src/features/donations/components/paymentMethods/CardPayments.tsx

```diff
@@ -101,7 +102,20 @@ function CardPayments({
   }, [CardNumberElement, CardExpiryElement, CardCvcElement]);
 
   const createPaymentMethodCC = (cardElement: any) => {
-    return stripe.createPaymentMethod('card', cardElement)
+    return stripe.createPaymentMethod({
+      type: 'card',
+      card: cardElement,
+      billing_details: {
+        name: `${donorDetails.firstname} ${donorDetails.lastname}`,
```

Comment:
Legacy payments fail here with the following error as probably the users name is not available (I only see the API call https://app-staging.plant-for-the-planet.org/public/v1.3/de/paymentInfo/JFCOX8DBEI - nothing loading a user name):
```
CardPayments.tsx:109 Uncaught (in promise) TypeError: Cannot read property 'firstname' of undefined
```
Tested with https://planet-webapp-kjunubc87.vercel.app/donation-payment/JFCOX8DBEI

Reply:
I did a donation in the Treecounter app without being logged in just entering a random address.

Reply:
donorDetails is not probably not a mandatory property for CardPayments - could such an error be prevented by type checking?

Reply:
This was added to pass billing details to stripe.

@sagararyal we will either have to remove the billing details from CC method or add the user's details in this API - `https://app-staging.plant-for-the-planet.org/public/v1.3/en/paymentInfo/JFCOX8DBEI`

Reply:
I have pushed a condition, if donor details are not present we don't pass the billing details

Reply:
Now the API call can be triggered.